src/foam/dao/index/Index.js

```diff
@@ -15,52 +15,114 @@
  * limitations under the License.
  */
 
-/* Indexed Memory-based DAO. */
+/**
+  The Index interface for an ordering, fast lookup, single value,
+  index multiplexer, or any other MDAO select() assistance class.
 
-/*
- * TODO:
- *  update(oldValue, newValue)
- *  reuse plans
- *  add ability for indices to pre-populate data
- */
+  Each Index subclass also defines an IndexNode class. Index defines
+  the structure of the index, including estimate() to gauge its
+  probable performance for a query, while IndexNode implements the
+  data nodes that hold the indexed items and plan and execute
+  queries. For any particular operational Index, there may be
+  many IndexNode instances:
+
+<pre>
+                 1---------> TreeIndex(id)
+  MDAO: AltIndex 2---------> TreeIndex(propA) ---> TreeIndex(id) -------------> ValueIndex
+        | 1x AltIndexNode    | 1x TreeIndexNode    | 14x TreeIndexNodes         | (DAO size)x ValueIndexNodes
+           (2 alt subindexes)     (14 nodes)             (each has 0-5 nodes)
+</pre>
+  The base AltIndex has two complete subindexes (each holds the entire DAO).
+  The The TreeIndex on property A has created one TreeIndexNode, holding one tree of 14 nodes.
+  Each tree node contains a tail instance of the next level down, thus
+  the TreeIndex on id has created 14 TreeIndexNodes. Each of those contains some number
+  of tree nodes, each holding one tail instance of the ValueIndex at the end of the chain.
 
-/** The Index interface for an ordering, fast lookup, single value,
-  index multiplexer, or any other MDAO select() assistance class. */
+*/
 foam.CLASS({
   package: 'foam.dao.index',
   name: 'Index',
 
+  properties: [
+    {
+      class: 'Class',
+      name: 'tailClass',
```

Comment:
Refactoring name change missed: should be 'nodeClass', docs added (it's the type of nodes this index creates in createNode())